sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/testing/TestDataflowPipelineRunner.java

```diff
@@ -220,10 +220,11 @@ public String toString() {
   private static class CancelWorkflowOnError implements JobMessagesHandler {
     private final DataflowPipelineJob job;
     private final JobMessagesHandler messageHandler;
-    private volatile String errorMessage = "";
+    private final StringBuffer errorMessage;
```

Comment:
Yes to concurrency.

This gets appended to in one thread and read from another thread otherwise I would have used stringbuilder.